### PR TITLE
Collect inode data on FreeBSD and add filesystem specs

### PR DESCRIPTION
**Pull request by tas50**
Modified version of the inode collection code from the Linux plugin.
Also add a filesystem spec for FreeBSD which is about 1/3 the specs from
the Linux filesystem spec, but modified for command output on FreeBSD

### CODE REVIEW

**Comment by mcquin** [comment]
:+1:

**Comment by mcquin** [comment]
\cc @chef/client-core

File: lib/ohai/plugins/freebsd/filesystem.rb [hunk 1]
```diff
@@ -39,6 +40,23 @@
       end
     end
 
+    # Grab filesystem inode data from df
+    so = shell_out("df -iP")
+    so.stdout.lines do |line|
+      case line
+      when /^Filesystem\s+Size/
```
Comment (btm) [inline lib/ohai/plugins/freebsd/filesystem.rb]: I don't see why we have this pattern everywhere, but you want `when /^Filesystem/` here.

File: lib/ohai/plugins/freebsd/filesystem.rb [hunk 2]
```diff
@@ -39,6 +40,23 @@
...
+        next
+      when /^(.+?)\s.*\d+\%\s+(\d+)\s+(\d+)\s+(\d+\%)\s+(.+)$/
```
Comment (btm) [inline lib/ohai/plugins/freebsd/filesystem.rb]: greedy regex is greedy. :p

**Comment by btm** [comment]
:+1:

File: lib/ohai/plugins/freebsd/filesystem.rb (hunk 1 above)
Comment (tas50) [inline lib/ohai/plugins/freebsd/filesystem.rb]: Total cargo cult from Linux here. I'll fix it

File: lib/ohai/plugins/freebsd/filesystem.rb (hunk 2 above)
Comment (tas50) [inline lib/ohai/plugins/freebsd/filesystem.rb]: Fixed

**Pull request merged**